### PR TITLE
Add Create User Role backend functionality

### PR DESCRIPTION
**Pull request by FirestarJes**
See title

### CODE REVIEW

File: source/Energinet.DataHub.MarketParticipant.Application/Commands/UserRoles/CreateUserRoleResponse.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.MarketParticipant.Domain.Model.Users;
+
+namespace Energinet.DataHub.MarketParticipant.Application.Commands.UserRoles
+{
+    public sealed record CreateUserRoleResponse(UserRoleId UserRoleId);
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Application/Commands/UserRoles/CreateUserRoleResponse.cs:19]: Do not return domain model to caller. Return Guid.

File: source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        _userRoleRepository = userRoleRepository;
+    }
+
+    public async Task<CreateUserRoleResponse> Handle(
+        CreateUserRoleCommand request,
+        CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(request);
+
+        var userRole = await _userRoleRepository
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs:43]: Create a domain model first, then send that to the repository for persistance.
`var userRole = new UserRole(...); await _userRoleRepository.AddAsync(userRole);`

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserRoleStatus.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Energinet.DataHub.MarketParticipant.Domain.Model.Users
+{
+    public enum UserRoleStatus
+    {
+        Active = 1,
+        InActive = 2,
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserRoleStatus.cs:20]: `Inactive`

File: source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleRepository.cs [hunk 4]
```diff
@@ -37,17 +38,16 @@ public async Task<IEnumerable<UserRole>> GetAllAsync()
...
+        var selectedUserRoleFunctions = queryable.SelectMany(e => e.EicFunctions.Select(r => new { r.EicFunction, r.UserRoleId, e.Id, e.Name, e.Description, e.Status }));
 
         var userRoles = selectedUserRoleFunctions
             .Select(r => new UserRole(
                 new UserRoleId(r.Id),
                 r.Name,
-                new List<EicFunction>(),
+                r.Description,
+                r.Status,
                 new List<Permission>(),
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleRepository.cs:49]: You should have the data/list on the entity, just need to assign it.

File: source/Energinet.DataHub.MarketParticipant.IntegrationTests/Repositories/UserRoleRepositoryTests.cs [hunk 5]
```diff
@@ -217,4 +217,30 @@ public async Task GetAsync_MultipleFunctions_DoesNotReturnWhenMissingFunction()
...
+    [Fact]
+    public async Task CreateAsync_AllValid_ReturnsUser()
+    {
+        // Arrange
+        await using var host = await OrganizationIntegrationTestHost.InitializeAsync(_fixture);
+        await using var scope = host.BeginScope();
+        await using var context = _fixture.DatabaseManager.CreateDbContext();
+        await using var context2 = _fixture.DatabaseManager.CreateDbContext();
+        var userRoleTemplateRepository = new UserRoleRepository(context);
+        var userRoleTemplateRepository2 = new UserRoleRepository(context);
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.IntegrationTests/Repositories/UserRoleRepositoryTests.cs:230]: Should be context2, me thinks.

File: source/Energinet.DataHub.MarketParticipant.Tests/Validation/CreateUserRoleCommandRuleSetTests.cs [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            Assert.Contains(propertyName, result.Errors.Select(x => x.PropertyName));
+        }
+
+        [Theory]
+        [InlineData("", false)]
+        [InlineData(null, false)]
+        [InlineData("  ", false)]
+        [InlineData(ValidName, true)]
+        [InlineData("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", true)]
+        [InlineData("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaX", false)]
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Tests/Validation/CreateUserRoleCommandRuleSetTests.cs:58]: https://www.youtube.com/watch?v=EGRAxjn_bzw

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserRoleStatus.cs (hunk 3 above)
Comment (xnetsrak) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserRoleStatus.cs:20]: no 0 ? just thinking that default value in db is 0 ?

File: source/Energinet.DataHub.MarketParticipant.IntegrationTests/Repositories/UserRoleRepositoryTests.cs (hunk 5 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.IntegrationTests/Repositories/UserRoleRepositoryTests.cs:230]: Fixed

File: source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleRepository.cs (hunk 4 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleRepository.cs:49]: Not quite sure i understand?

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserRoleStatus.cs (hunk 3 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserRoleStatus.cs:20]: Updated values to be 0 and 1 instead, fixed name

File: source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs (hunk 2 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs:43]: We don't do this in other handlers, i.e. CreateGridAreaHandler or CreateOrganizationHandler, so perhaps we should align all of them?

File: source/Energinet.DataHub.MarketParticipant.Application/Commands/UserRoles/CreateUserRoleResponse.cs (hunk 1 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Application/Commands/UserRoles/CreateUserRoleResponse.cs:19]: Fixed

File: source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs (hunk 2 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs]: Ahh, we actually do that in the servicefactory class, is it overkill to create one for UserRole too?

Comment (defectiveAi) [reply source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs]: We do it for GridArea and Organization too, but the construction is more complicated and so is happening in a domain factory service instead. But the objects are still created/validated in the domain layer, and sent to repository for persistance only.

Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs]: Yes i noticed, i'm fixing it now

File: source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleRepository.cs (hunk 4 above)
Comment (defectiveAi) [reply source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleRepository.cs:49]: Actually, I think you can just write this method as:
```
public async Task<IEnumerable<UserRole>> GetAllAsync()
{
    var userRoles = await BuildUserRoleQuery().ToListAsync().ConfigureAwait(false);
    return userRoles.Select(MapUserRole);
}

Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/Repositories/UserRoleRepository.cs:49]: Fixed

File: source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs (hunk 2 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Application/Handlers/UserRoles/CreateUserRoleHandler.cs]: Should be fixed now

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**